draft-nygren-httpbis-httpssvc.md

```diff
@@ -0,0 +1,791 @@
+   address queries.
+1. Wait for the HTTPSSVC record response.
+1. If the response is nonempty, proceed with alternative service
+   connection and ignore the address query responses.
+1. Otherwise, wait for the address queries and connect as usual.
+
+Note that this process is also expected to be faster than Alt-Svc over
+HTTP in the case of HTTP Opportunistic Upgrade Probing (Section 2 of
+{{?RFC8164}}).
+
+
+# DNS Server Behaviors
+
+Recursive DNS servers SHOULD resolve SvcDomainName records and include
+them in the Additional Section (along with any relevant CNAME
+records).  For SvcRecordType=0, recursive DNS servers should attempt
+to resolve and include A, AAAA, and HTTPSSVC records.  For
+SvcRecordType=1, recursive DNS servers should attempt to resolve and
+include A and AAAA records.
+
+Authoritative DNS servers SHOULD return A, AAAA, and HTTPSSVC records
+(as well as any relevant CNAME records) in the Additional Section for
+any in-bailiwick SvcDomainNames.
+
+
+# Extensions to enhance privacy
+
+## Alt-Svc parameter for cleartext SNI value
+
+An Alt-Svc "sni" parameter is defined for specifying
+the cleartext SNI value to be sent when connecting to this
+alternative service.  When present, clients SHOULD
+send this SNI value when connecting to the alternative.
+
+For example:
+
+ sni="_wildcard.example.com"
+
+would indicate that "_wildcard.example.com" should be sent
+when connecting to the corresponding alternative service.
+For a large set of origin names covered by a wildcard cert,
+this can help hide which specific origin name is needed.
+
+Because service operators control the HTTPSSVC record,
+this allows them to specify what information they need
+from the SNI value.
+
+Servers MUST still be prepared for clients to continue to send
+an SNI value of the origin hostname.
+
+This parameter MAY also be sent in Alt-Svc HTTP response
+headers and HTTP/2 ALTSVC frames.
+
+
+## Alt-Svc parameter for ESNI keys
+
+An Alt-Svc "esnikeys" parameter is defined for specifying
+ESNI keys corresponding to an alternative service.
+The value of the parameter is an ESNIKeys structure {{!ESNI}}
+encoded in {{!base64=RFC4648}}.
+
+This parameter MAY also be sent in Alt-Svc HTTP response
+headers and HTTP/2 ALTSVC frames.
+
+
+## Interaction with other standards
+
+The purpose of this standard is to reduce connection latency and
+improve user privacy.  Server operators implementing this standard
+SHOULD also implement TLS 1.3 {{!I-D.ietf-tls-tls13}} and OCSP Stapling
+{{!RFC6066}}, both of which confer substantial performance and privacy
+benefits when used in combination with HTTPSSVC records.
+
+To realize the greatest privacy benefits, this proposal is intended for
+use with a privacy-preserving DNS transport (like DNS over TLS
+{{!RFC7858}} or DNS over HTTPS {{!DOH=I-D.ietf-doh-dns-over-https}}),
+and with the "SNI" Alt-Svc Parameter
+{{!AltSvcSNI=I-D.bishop-httpbis-sni-altsvc}}.  However, performance
```

Comment:
This references the draft which defines sni=, but you redefine it in 6.1.  Need to resolve.

Reply:
As I've mentioned before, I tend to think we should drop this, based on the chilly reception we've gotten thus far.  The browser folks I've talked to seemed particularly opposed to implementing any protocol that created mismatch between the SNI and the actual validation process, because of the potential for security bugs.

Reply:
I'm amenable to leaving them separate -- my comment is purely around document consistency.  It would be consistent if all references to both were removed, too.  :-)

Reply:
I'm fine with dropping this.  It could always be a separate document and reduces complexity here.

Reply:
Created PR #17 for this.